Polyhedron/demo/Polyhedron/MainWindow.cpp

```diff
@@ -3591,6 +3592,44 @@ void MainWindow::on_action_Save_triggered()
   }
 }
 
+void MainWindow::test_all_actions()
+{
+  int nb_items = scene->numberOfEntries();
+  selectSceneItem(0);
+  Q_FOREACH(PluginNamePair pnp, plugins)
+  {
+    Polyhedron_demo_plugin_interface* plugin = pnp.first;
+    Q_FOREACH(QAction* action, plugin->actions()){
+      if(plugin->applicable(action)){
+        qDebug()<<"Testing "<<pnp.second<<"and "<<action->text()<<" on";
+        qDebug()<<scene->item(scene->mainSelectionIndex())->name()<<"...";
+        action->triggered();
+        if(isLocked())
+        {
+          getMutex()->lock();
+          getWaitCondition()->wait(getMutex());
+          getMutex()->unlock();
+          //get the "done event" that add items after the meshing thread is finished to execute before we start the next action.
+          QCoreApplication::processEvents();
+        }
```

Comment:
That looks wrong, but I am not completely sure.

The access (reads and writes) to the `locked` boolean should be protected by the mutex.

See https://www.modernescpp.com/index.php/c-core-guidelines-be-aware-of-the-traps-of-condition-variables for example.

Reply:
your right, it was the cas eat some point of the dev. I didn't see it wasn't anymore, it's a mistake.